src/data/matrix/basic.lean

```diff
@@ -59,9 +59,19 @@ by { ext, simp, }
 /-- The transpose of a matrix. -/
 def transpose (M : matrix m n α) : matrix n m α
 | x y := M y x
-
 localized "postfix `ᵀ`:1500 := matrix.transpose" in matrix
 
+/-- The conjugate transpose of a matrix. -/
+def conj_transpose [has_star α] (M : matrix m n α) : matrix n m α :=
+  M.transpose.map star
+localized "postfix `ᴴ`:1500 := matrix.conj_transpose" in matrix
+
+instance [has_star α] : has_star (matrix n n α) := {star := conj_transpose}
+
+lemma star_eq_conj_transpose [has_star α] (M : matrix m m α) : star M = Mᴴ := rfl
+
+@[simp] lemma star_apply [has_star α] (M : matrix n n α) (i j) : (star M) i j = star (M j i) := rfl
```

Comment:
IMO these lemmas made more sense where you had them before, after all the `conj_transpose` lemmas. As you say, the `star` structure on square matrices is only really a special case.